modules/core/shared/src/main/scala/lucuma/core/math/WavelengthCoverage.scala

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.math
+
+import algebra.instances.all.given
+import cats.Order
+import cats.Show
+import coulomb.Quantity
+import coulomb.syntax.*
+import eu.timepit.refined.auto.*
+import eu.timepit.refined.cats.*
+import eu.timepit.refined.numeric.*
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import eu.timepit.refined.types.numeric.PosInt
+import lucuma.core.math.units.Angstrom
+import lucuma.core.math.units.Micrometer
+import lucuma.core.math.units.Nanometer
+import lucuma.core.math.units.Picometer
+import lucuma.core.optics.Format
+import monocle.Iso
+import monocle.Prism
+
+import java.math.RoundingMode
+import scala.annotation.targetName
+
+/**
+ * Represents a span of the spectrum, represented as a small positive wavelength
+ * delta in picometers. It can be "anchored" by providing a starting, ending, or
+ * central `Wavelength`, which results in a wavelength interval (ie: a 
+ * `BoundedInterval[Wavelength]`).
+ */
+opaque type WavelengthCoverage = Quantity[PosInt, Picometer]
+
+object WavelengthCoverage:
+  extension (w: WavelengthCoverage) {
+    def toPicometers: Quantity[PosInt, Picometer] =
+      w
+
+    /**
+     * Alias for `toPicometers`.
+     */
+    def pm: Quantity[PosInt, Picometer] =
+      toPicometers
+
+    /**
+      * Create a wavelength range centered on the given value. Will be truncated at 
+      * Wavelength.Min and Wavelength.Max.
+      */
+    def centeredAt(λ: Wavelength): BoundedInterval[Wavelength] = 
```

Comment:
I assume planning to use this class to calculate `coverageInterval` in the SpectroscopyModesMatrix in explore? It seems there we have another case where the interval gets `clipped` within a larger interval. So given a central wavelength, the coverage interval can't extend beyond the capabilities of the configuration. Do you want to include that case here, or just handle it manually in SpectroscopyModesMatrix?

Reply:
> Do you want to include that case here, or just handle it manually in SpectroscopyModesMatrix?

It's done via `BoundedInterval.intersect` in `SpectroscopyModesMatrix`.